src/qibo/models/circuit.py

```diff
@@ -1012,6 +1024,41 @@ def __call__(self, initial_state=None, nshots=None):
         """Equivalent to ``circuit.execute``."""
         return self.execute(initial_state=initial_state, nshots=nshots)
 
+    @property
+    def raw(self) -> dict:
+        """Serialize to dictionary.
+
+        This is a thin wrapper over :meth:`Gate.raw`.
+        """
+        return {
+            "queue": [gate.raw for gate in self.queue],
+            "nqubits": self.nqubits,
+            "density_matrix": self.density_matrix,
+            "qibo_version": qibo.__version__,
+        }
+
+    @classmethod
+    def load(cls, raw):
+        """Load from serialization.
+
+        Essentially the counter-part of :meth:`raw`.
+        """
+        circ = cls(raw["nqubits"], density_matrix=raw["density_matrix"])
+
+        for gate in raw["queue"]:
+            circ.add(Gate.load(gate))
+
+        return circ
+
+    def dump_json(self, stream):
```

Comment:
I would simply call `dump` and pass the `filename` instead of `stream`.

Reply:
I put `_json` on purpose, to be explicit about the action performed.

`json.dump` itself is dumping on a file descriptor or a stream, so `filename` would be plainly wrong.
This has to be used as:
```py
with open("my_file.json", "w") as fd:
    circuit.dump_json(fd)
```

Using filenames is uselessly less general, because it prevents you to use in-memory streams to emulate files (as those available in `io`).